### ISSUE TITLE
Switch to gtk3-4

### ISSUE THREAD

**Issue by Bundy01** [calf-studio-gear/calf#328]
Hi,

Do you plan to change the dependency from gtk2 to gtk3-4?

Regards.

**Comment by suedwestlicht** [comment]
I am afraid that wouldn't help since there will always be an incompatible gtk(n+1) where the main problem is that it can't be statically linked.

So a switch to another toolkit will be neccessary.

**Comment by enorrmann** [comment]
is this the reason why plugins are not working with ardour anymore ?

**Comment by suedwestlicht** [comment]
> is this the reason why plugins are not working with ardour anymore ?

No. Ardour uses a modified version of gtk2 that's incompatible with standard gtk2.
CALF plugins are still working fine in Qtractor.

**Comment by trebmuh** [comment]
> Or is GTK3 not wanted due to plugin compatibility issues with GTK2 hosts?

See https://github.com/calf-studio-gear/calf/issues/346

**Comment by juzzlin** [comment]
> It sounds almost like a job for AI to port the code to GTK3. Does anyone have experience with that?
> 
> Or is GTK3 not wanted due to plugin compatibility issues with GTK2 hosts?

Gemini CLI with Gemini 3 could probably easily do that. I have used Gemini CLI to implement parts of my Noteahead MIDI tracker and it's incredible.

**Comment by RavenRandomz** [comment]
> > It sounds almost like a job for AI to port the code to GTK3. Does anyone have experience with that?
> > Or is GTK3 not wanted due to plugin compatibility issues with GTK2 hosts?
> 
> Gemini CLI with Gemini 3 could probably easily do that. I have used Gemini CLI to implement parts of my Noteahead MIDI tracker and it's incredible.

Important thing for your project and this project: You cannot copyright AI generated code. It means that you cannot enforce any license for any portion of the code that is AI generated.

**Comment by juzzlin** [comment]
> > > It sounds almost like a job for AI to port the code to GTK3. Does anyone have experience with that?
> > > Or is GTK3 not wanted due to plugin compatibility issues with GTK2 hosts?
> > 
> > 
> > Gemini CLI with Gemini 3 could probably easily do that. I have used Gemini CLI to implement parts of my Noteahead MIDI tracker and it's incredible.
> 
> Important thing for your project and this project: You cannot copyright AI generated code. It means that you cannot enforce any license for any portion of the code that is AI generated.

AFAIK it's perfectly fine to integrate and copyright snippets created by AI. It's considered to be a tool. Generating the whole software in one go is a different thing.
